lib/modules/manager/circleci/schema.ts

```diff
@@ -4,14 +4,30 @@ export const CircleCiDocker = z.object({
   image: z.string(),
 });
 
-export type CircleCiJob = z.infer<typeof CircleCiJob>;
 export const CircleCiJob = z.object({
   docker: z.array(CircleCiDocker).optional(),
 });
+export type CircleCiJob = z.infer<typeof CircleCiJob>;
+
+type Orb = {
+  executors?: Record<string, CircleCiJob>;
+  jobs?: Record<string, CircleCiJob>;
+  orbs?: Record<string, string | Orb>;
+};
```

Comment:
oh I see this recursive type 🫣

@zharinov do you know how to declare this properly with zod? I don't like the mix of zod and normal type, it will cause runtime issues later.

Reply:
unless I'm misunderstanding, I don't think it's supported? https://github.com/colinhacks/zod/issues/3331

Reply:
it's possible 

https://github.com/colinhacks/zod?tab=readme-ov-file#recursive-types

Reply:
sure, but if I understand correctly that's using `z.infer` to construct the concrete type. i'm fine making this change, it's just a bit more verbose IMO
```typescript
// eslint-disable-next-line
const baseOrb = z.object({
  executors: z.record(z.string(), CircleCiJob).optional(),
  jobs: z.record(z.string(), CircleCiJob).optional(),
});

type Orb = z.infer<typeof baseOrb> & {
  orbs?: Record<string, string | Orb>;
};
```

Reply:
sure, but the base type can then be used later for the real type again, so less duplicate code which needs to match types.